### PR TITLE
Fix issue #3737

### PR DESCRIPTION
**Pull request by alan-j-hu**
As @emillon bisected, the bug was introduced in https://github.com/ocaml/dune/commit/5c133a06f8c5f252abe2b7a855834530fd57d5cb. This commit rewrote `pump_events` as `iter`. However, while `pump_events` ran `Console.Status_line.set (Fun.const None)` after the jobs were done, the rewritten code only did so before the jobs were started.

I assume that it is okay to remove the call `Console.Status_line.set (Fun.const None)` that occurs before the jobs start running, but I'm not sure because I'm unfamiliar with the Dune codebase.

### CODE REVIEW

**Comment by emillon** [comment]
That seems to fix it, thanks! I can't comment on the implementation, but the effect seems to be there.

File: src/dune_engine/scheduler.ml [hunk 1]
```diff
@@ -702,10 +702,10 @@ end = struct
     let fiber =
       Fiber.Var.set t_var t (fun () -> Fiber.with_error_handler f ~on_error)
     in
-    Console.Status_line.set (Fun.const None);
     match Fiber.run fiber ~iter with
     | res ->
       assert (Event.pending_jobs () = 0);
+      Console.Status_line.set (Fun.const None);
```
Comment (jeremiedimino) [inline src/dune_engine/scheduler.ml:708]: That seems about right, but I feel like we should do this cleanup systematically, i.e. not just in the case of a successful termination but also in the case `exception Abort err | exception exn`

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/dune_engine/scheduler.ml (hunk 1 above)
Comment (dosaylazy) [reply src/dune_engine/scheduler.ml:708]: Okay! Should I use `Fun.protect` for this, or should I just copy over the line to the other `match` branches?

**Comment by dosaylazy** [comment]
@jeremiedimino Okay, I made the changes.

**Review by ghost — APPROVED** [review]
Looks good. I used a more specific variable name and added a changelog entry. Good to merge once CI is green.

**Comment by jeremiedimino** [comment]
I looked into why the bootstrap suffers from the same problem: it's because with some version of OCaml, `at_exit` functions don't work with`-output-complete-exe` and we build the bootstrap build system (`boot/duneboot.ml`) with `-output-complete-exe`.

**Comment by jeremiedimino** [comment]
Merged, thanks @dosaylazy for the debugging and fix!

**Pull request merged**